sqliterepo/cache.c

```diff
@@ -105,6 +106,30 @@ struct sqlx_cache_s
 
 /* ------------------------------------------------------------------------- */
 
+static gboolean
+_ram_exhausted(void)
+{
+#if HAVE_MALLINFO
+	/* JFS: check on the glibc level. It would not detect a memory exhaust
+	 * if another library would be in use (e.g. jemalloc, dietlib, etc). But
+	 * it causes no syscall and currently matches our usage on the field
+	 * (where the glibc is used). */
+	if (sqliterepo_max_rss > 0) {
+		struct mallinfo mi = mallinfo();
+		gint64 total = 0;
+		total += mi.arena;
+		total += mi.ordblks;
+		total += mi.smblks;
+		total += mi.hblkhd;
```

Comment:
`arena` and `hblkhd` should be sufficient ("The total amount of memory allocated by means other than mmap" and "The number of bytes in blocks currently allocated using mmap").